code/game/objects/items/item_arms.dm

```diff
@@ -0,0 +1,115 @@
+/obj/item/item_arm
+	name = "prosthetic item template"
+	desc = "A piece of equipment that replaced your arm"
+	item_flags = ABSTRACT | DROPDEL
+	w_class = WEIGHT_CLASS_HUGE
+	var/obj/droped_object
+
+/obj/item/item_arm/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, HAND_REPLACEMENT_TRAIT)
+
+/obj/item/item_arm/Destroy()
+	var/obj/item/bodypart/part
+	new droped_object(get_turf(src))
+	if(iscarbon(loc))
+		var/mob/living/carbon/holder = loc
+		var/index = holder.get_held_index_of_item(src)
+		if(index)
+			part = holder.hand_bodyparts[index]
+	. = ..()
+	if(part)
+		part.drop_limb()
+
+/obj/item/shock_arm_unattached
+	name = "Unattached Shock Arm"
+	desc = "Shock arm ready to be surgicaly attached"
+	icon = 'icons/obj/item_arms.dmi'
+	icon_state = "shock_arm"
+
+/obj/item/item_arm/shock_arm
+	name = "Shock Arm"
+	desc = "Device capable of incapiciating oponents at range using low intensity electricity arcs."
+	force = 0
+	reach = 5
+	icon = 'icons/obj/item_arms.dmi'
+	icon_state = "shock_arm"
+	item_state = "shock_arm"
+	lefthand_file = 'icons/mob/inhands/weapons/item_arm_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/item_arm_righthand.dmi'
+	droped_object = /obj/item/shock_arm_unattached
+	var/shock_range = 5
+	var/shock_stamina = 15
+	var/mob/living/shock_target
+	var/mob/living/shock_instigator
+	var/is_shocking = TRUE
```

Comment:
`is_shocking` doesn't do anything, it never gets called it only gets set to either `TRUE` or `FALSE`

Reply:
Thank you for taking your time to check my code and point this out